### ISSUE TITLE
qubes-manager: "reset to defaults" option

### ISSUE THREAD

**Issue by axon-qubes** [QubesOS/qubes-issues#1363]
It's standard in most GUI program to have a button which resets all options to their default settings, where "default" means roughly "how things would look on a fresh install."

**Comment by marmarek** [comment]
What exactly do you have in mind? "Reset VM settings"? Or something more
global?

-- 
Best Regards,
Marek Marczykowski-Górecki
Invisible Things Lab
A: Because it messes up the order in which people normally read text.
Q: Why is top-posting such a bad thing?

**Comment by axon-qubes** [comment]
Qubes Manager --> System --> Qubes Global Settings

In this window, the user can change things like: minimal VM memory, dom0 memory boost, whether to keep DispVMs in memory, whether to check for dom0 updates, and whether to check for VM updates. A user might change some of those settings (being new to Qubes, not realizing that her custom choice is a bad idea), then later realize that she wants to revert to the defaults. But maybe she doesn't remember what the defaults are. It would be nice to have a button which sets everything to the defaults.

**Comment by fepitre** [comment]
I would add something related to this: resetting default applications list in VM settings. @marmarta @marmarek do you think that would be reasonable to do?

**Comment by airelemental** [comment]
The default applications list is "show every application", so can't this be accomplished by clicking the ">>"?

The only settings reset that I think would be useful, is resetting the firewall so that it clears any qvm-firewall modifications, allowing the firewall tab to be usable again.

**Comment by andrewdavidwong** [comment]
> The default applications list is "show every application", so can't this be accomplished by clicking the ">>"?

Are you sure? I thought the default was just a handful of common applications.

> One specific settings reset that might be useful, is resetting the firewall so it clears any `qvm-firewall` CLI modifications, allowing the firewall tab to be usable again.

This is a separate issue: #4710

**Comment by fepitre** [comment]
> > The default applications list is "show every application", so can't this be accomplished by clicking the ">>"?
> 
> Are you sure? I thought the default was just a handful of common applications.
He is wrong and you are right. The default application list is what we provide in every builder-*/template_scripts/appmenus_*

**Comment by marmarta** [comment]
@marmarek, what is the source of global defaults? or is there no simple answer to this one?

**Comment by marmarek** [comment]
Those values being just properties of `Qubes()` object, can be reset by assigning `qubesadmin.DEFAULT` value - you can also try getting the default using `property_get_default()` method. But it may not work for some, like `default_template` which is basically the first template installed on the system that may not be present anymore at a later time. Similar case for default kernel. Some others have default value `None`, only set later by initial setup wizard (this is the case of default_netvm for example - which BTW really depends on the choice made in the initial-setup).
Settings modified directly in some config file (like qmemman parameters) don't have programmatic way of getting default values. In case of qmemman, removing them from the file will make it use defaults, but you won't know the actual value, unless you copy it from the qmemman source code...

In short: there is no simple way to get "default" values (understood as "values just after installation") for global settings. You can get some of them by copying logic that set them during installation, but for some it may still not be enough (like default template, which could be removed since then).

This is much easier for VM properties, where default values for most properties are actually meaningful, especially for "normal" AppVMs. There are still few corner cases - like 'netvm' property of a sys-net itself, but those applies to system VMs mostly. Those things again are set by the initial setup (through salt).
